.github/workflows/cli_core_unit_test_mac.yaml

```
@@ -0,0 +1,56 @@
+name: Tanzu CLI Unit Tests on MacOS
+
+on:
+  pull_request:
+    branches: [main, release-*]
+  push:
+    branches: [ main, test*, release-*]
+
+jobs:
+  build:
+    name: Tanzu CLI Unit Tests
+    runs-on: macos-latest
```

You should not required to create a new unit test job to run the same thing on a different OS. You can use the `strategy` field to specify this. https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-multi-dimension-matrix

You should be able to do something like the below in the `cli_core_unit_test.yaml` file to run the pipeline on both OS.

```suggestion
    strategy:
      matrix:
        os: [macos-latest, ubuntu-latest]
    runs-on: ${{ matrix.os }}
```

---

thanks for suggestion, fixed.